x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/components/analytics_tables/inline_tags_list.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiBadgeGroup, EuiBadge, EuiToolTip } from '@elastic/eui';
+
+import { Query } from '../../types';
+
+interface Props {
+  tags: Query['tags'];
```

Comment:
This is fine, just asking out of curiosity. Why use `Query['tags']` here instead of just `string[]`.

Reply:
@byronhulcher got me on this habit a while ago! https://github.com/elastic/kibana/pull/88095#discussion_r556640406

I like the idea of it because it means if we ever change a type/data structure in the future we can update it in one place and all other places that we're just passing that same prop to get it automatically.

Reply:
It's also more explicit about what specifically should be passed, which helps with understanding I think.